### PR TITLE
Translation blog

### LINKED ISSUE #183: Translation blog
fix arabic translation to blog page

### PR DESCRIPTION
**Pull request by mohammad-jundi**
Added translations to the blog section

### CODE REVIEW

File: src/components/Blogs/index.jsx [hunk 1]
```diff
@@ -3,24 +3,40 @@ import { Container, Row } from "react-bootstrap";
...
+    };
+
     fetchBlogPosts();
-  }, []);
+  }, [languageRef]);
+
+  i18next.on("languageChanged", (lng) => {
+    let ref = i18next.translator.language;
+    if (ref === "en") {
+      setLanguageRef("10221");
```
Comment (louisrli) [inline src/components/Blogs/index.jsx]: magic string 

please google / read technical manual

what does 10221 mean?
what does 38229 mean?

pretend a teammate just joined and has to read this code. would they understand?

Comment (Jamil-Bailony) [reply src/components/Blogs/index.jsx]: I was looking at this weird number :)

Comment (mohammad-jundi) [reply src/components/Blogs/index.jsx]: fixed

File: src/components/Blogs/index.jsx [hunk 2]
```diff
@@ -3,24 +3,42 @@ import { Container, Row } from "react-bootstrap";
 import SearchBar from "../SearchBar";
 import BlogCard from "../BlogCard";
 import "./index.scss";
+import i18next from "i18next";
 
-const POSTS_PER_PAGE = 9;
+const POSTS_PER_PAGE = 18;
 const BLOG_API_URL = `https://public-api.wordpress.com/wp/v2/sites/safeplace102505649.wordpress.com/posts?per_page=${POSTS_PER_PAGE}`;
+const EN_REFERENCE_NUMBER = "10221";
```
Comment (louisrli) [inline src/components/Blogs/index.jsx]: still i dont know what this means. again imagine the scenario, i am a new teammate. use some comments. explain

**Review by Jamil-Bailony — APPROVED** [review]

File: src/components/Blogs/index.jsx [hunk 3]
```diff
@@ -3,24 +3,45 @@ import { Container, Row } from "react-bootstrap";
...
+const AR_REFERENCE_NUMBER = "38299";
+// numbers "10221" and "38299" are the refernce numbers from word press that we get after taging each post to its language in order to make the change between the languages
```
Comment (louisrli) [inline src/components/Blogs/index.jsx]: comments go above what they are explaining.

you dont need line 13/14, delete those

Comment (jennetturkkan) [reply src/components/Blogs/index.jsx]: can you please take a look now? @louisrli

File: src/components/Blogs/index.jsx [hunk 4]
```diff
@@ -3,24 +3,43 @@ import { Container, Row } from "react-bootstrap";
...
   const [blogPosts, setBlogPosts] = useState([]);
   const [blogsForSearch, setBlogsForSearch] = useState([]);
-
-  const fetchBlogPosts = async () => {
-    const data = await fetch(BLOG_API_URL);
-    const fetchedBlogPosts = await data.json();
-    setBlogsForSearch(fetchedBlogPosts);
-    setBlogPosts(fetchedBlogPosts);
-  };
+  const [languageRef, setLanguageRef] = useState("10221");
```
Comment (louisrli) [inline src/components/Blogs/index.jsx]: fix

Comment (jennetturkkan) [reply src/components/Blogs/index.jsx]: fixed

File: src/components/Blogs/index.jsx [hunk 5]
```diff
@@ -3,24 +3,43 @@ import { Container, Row } from "react-bootstrap";
...
+    const fetchBlogPosts = async () => {
+      const data = await fetch(BLOG_API_URL);
+      const fetchedBlogPosts = await data.json();
+      const filteredPosts = await fetchedBlogPosts.filter((post) => {
+        return post.tags.join() === languageRef;
+      });
+
+      setBlogsForSearch(fetchedBlogPosts);
+      setBlogPosts(fetchedBlogPosts);
+      setBlogPosts(filteredPosts);
```
Comment (louisrli) [inline src/components/Blogs/index.jsx]: fix

Comment (jennetturkkan) [reply src/components/Blogs/index.jsx]: fixed

**Review by louisrli — APPROVED** [review]

**Comment by jennetturkkan** [comment]
@Ammar-64

**Review by louisrli — CHANGES_REQUESTED** [review]
see my comment on other PR

**Comment by louisrli** [comment]
also you need to move the reference numbers to a common file, instead of copy-pasting it between two files

**Comment by jennetturkkan** [comment]
is this how you want translation to be implemented? can you please take a look? @louisrli

**Comment by jennetturkkan** [comment]
@Ammar-64 don't merge it yet

File: src/components/Blogs/index.jsx [hunk 6]
```diff
@@ -3,24 +3,38 @@ import { Container, Row } from "react-bootstrap";
...
+    let ref = i18next.translator.language;
+    if (ref === "en") {
+      setLanguageRef(EN_REFERENCE_NUMBER);
+    } else if (ref === "ar") {
+      setLanguageRef(AR_REFERENCE_NUMBER);
+    }
+  });
 
   useEffect(() => {
+    const blogsApiUrl = `https://public-api.wordpress.com/wp/v2/sites/safeplace102505649.wordpress.com/posts?tags=${languageRef}`;
```
Comment (louisrli) [inline src/components/Blogs/index.jsx]: move the prefix back to BLOG_API_URL, add ${BLOG_API_URL}?tags=${languageRef}

**Review by louisrli — COMMENTED** [review]
otherwise correct and refs should be in common file, probably BLOG_API_URL too. no copy pasting

File: src/components/Blogs/index.jsx (hunk 6 above)
Comment (jennetturkkan) [reply src/components/Blogs/index.jsx]: ${BLOG_API_URL}?tags=${languageRef} this should be in a separate variable?

**Comment by jennetturkkan** [comment]
@louisrli please take a look

**Review by louisrli — APPROVED** [review]

**Comment by jennetturkkan** [comment]
@Ammar-64 now you can merge :)

**Pull request merged**